core/src/com/unciv/models/ruleset/unit/BaseUnit.kt

```diff
@@ -593,15 +590,28 @@ class BaseUnit : RulesetObject(), INonPerpetualConstruction {
             power += 4000
 
         // Uniques
-        for (unique in uniqueObjects) {
+        val allUniques = uniqueObjects.asSequence() +
+            promotions.asSequence()
+                .mapNotNull { ruleset.unitPromotions[it] }
+                .flatMap { it.uniqueObjects }
 
+        for (unique in allUniques) {
             when {
                 unique.isOfType(UniqueType.Strength) && unique.params[0].toInt() > 0 -> {
-                    if (unique.conditionals.any { it.isOfType(UniqueType.ConditionalVsCity) })
-                        power += (power * unique.params[0].toInt()) / 200
-                    else if (unique.conditionals.any { it.isOfType(UniqueType.ConditionalVsUnits) })
-                        power += (power * unique.params[0].toInt()) / 400
+                    power += (power * unique.params[0].toInt()) / when {
```

Comment:
According to my math, this is actually equivalent to `power *= unique.params[0].toPercent / 2f`, which looks more readable in my opinion. I've also changed this in #5361, so these might conflict because of that. Otherwise, looks fine :+1:

Reply:
I only touched that because Studio had some squigglies on that 'when'... I try to make one inspections check for each touched file after I'm done, not always treating each complaint, but at least considering each. We still have that bunch of 8-12 warnings actually appearing in github checks, but those I left at type 'unused parameter' only.